### PR TITLE
removed colors, added emoji support

### PR DESCRIPTION
**Pull request by jatraiz**
@dostrander @ateliercw

### CODE REVIEW

File: Pod/Classes/Logging/Log.swift [hunk 1]
```diff
@@ -61,59 +59,78 @@ open class Log {
             case .off: return "Disabled"
             }
         }
+
+        func emoji() -> String {
```
Comment (dostrander) [inline Pod/Classes/Logging/Log.swift]: #pp var

File: Pod/Classes/Logging/Log.swift [hunk 2]
```diff
@@ -61,59 +59,78 @@ open class Log {
...
+            }
+        }
     }
-    
+
     /// The log level, defaults to .Off
     open static var logLevel: Level = .off
     
     /// If true, prints text in color in accordance with the Xcode log color plug-in.
     open static var printColoredText: Bool = false
```
Comment (dostrander) [inline Pod/Classes/Logging/Log.swift]: #req `printEmojiedText` is that even a word?!?

Comment (dostrander) [inline Pod/Classes/Logging/Log.swift]: Oh wait I see the other one. Just remove this then

File: Pod/Classes/Logging/Log.swift [hunk 3]
```diff
@@ -61,59 +59,78 @@ open class Log {
...
-    
-    fileprivate struct Color {
-        static let red = Cmd.esc + "fg240,0,0;"
-        static let yellow = Cmd.esc + "fg200,200,0;"
-        static let green = Cmd.esc + "fg0,150,0;"
+
+    /// Date formatter for log
+    fileprivate static let dateformatter: DateFormatter = {
+        let df = DateFormatter()
+        df.dateFormat = "Y-MM-dd H:m:ss.SSSS"
```
Comment (dostrander) [inline Pod/Classes/Logging/Log.swift]: 👍 THANK YOU!

File: Pod/Classes/Logging/Log.swift [hunk 4]
```diff
@@ -61,59 +59,78 @@ open class Log {
...
+            print("\(levelString)\(date) \(objectName) \(functionName) line \(line):\n\(object())\n")
         }
     }
-    
+
     // MARK: Log Methods
     
-    static open func error<T>(_ object: T, _ fileName: String = #file, _ functionName: String = #function, _ line: Int = #line) {
-        log(object, level:.error, color: Color.red, fileName, functionName, line)
+    open static func error<T>(_ object: @autoclosure () -> T, _ fileName: String = #file, _ functionName: String = #function, _ line: Int = #line) {
```
Comment (dostrander) [inline Pod/Classes/Logging/Log.swift]: Nice idea with the `autoclosure`

Comment (dostrander) [inline Pod/Classes/Logging/Log.swift]: #? Do we want these methods `open` anymore?  I would think final would be more appropriate but I don't know why we made them open to begin with.

Comment (jatraiz) [inline Pod/Classes/Logging/Log.swift]: yeah, I'm not positive about using `open`, i just left that there b/c of the previous version. @nbonatsakis, any thoughts?

File: Pod/Classes/Logging/Log.swift (hunk 3 above)
Comment (ateliercw) [inline Pod/Classes/Logging/Log.swift]: implementes #32

File: Pod/Classes/Logging/Log.swift (hunk 4 above)
Comment (ateliercw) [inline Pod/Classes/Logging/Log.swift]: +1 for making them `public` instead of `open`, my guess is they were made extra permissive by the swift migrator

**Comment by ateliercw** [comment]
#? if we're making changes to this file, do we also want to match the log levels to those available in [OSLogType](https://developer.apple.com/reference/os/oslogtype) for easier future integration with os_log?

**Comment by dostrander** [comment]
@ateliercw IMO i think it'd be better to have a full integration over to `os_log` in one PR than having some of it part of this one.

**Pull request merged**